docarray/typing/tensor/abstract_tensor.py

```diff
@@ -136,4 +135,11 @@ def to_protobuf(self) -> 'NdArrayProto':
 
     def unwrap(self):
         """Return the native tensor object that this DocArray tensor wraps."""
+
+    @abc.abstractmethod
+    def __docarray_to_json_compatible__(self):
```

Comment:
how would you name it ? Maybe we should rename all of the `__docarray_xxx__` to `_docarray_xxx_`

Reply:
does it even have to be private? I am also not sure that the docarray namespacing is needed in all cases. why not just `to_json_compatible`?

Reply:
I believe private is needed here because the json compatible object is actually only compatible with orjson. So it is only a valid name under the docarray realm

Reply:
then lets go `_docarray_to_json_compatible` and I will rename all of the `__docarray_xxx___` to `_docarray_xxx` in another PR

Reply:
If tensor object where normal object we would have use private method without the `_docarray` namespace at the beginning. The problem here is that the tensor inherit from classes that we don't control (torch.Tensor, numpy.ndarray) and even worse it will inherit from classes that we don't know yet about (jax, tensorflow, another big deep learning framework that might not yet exist). And we only need of this backend to define a private method with the same name and we are screwed.  Adding the `_docarray` namesapce is a safe way to prevent this problem in the future.

Realm here is just a term I am making up it just mean inside our codebase

Reply:
in a nutshell: multiple inheritance from class that we don't control is dangerous and without a proper namespace things can colide so we use it preventively